### PR TITLE
Mapping streamline

### PR DESCRIPTION
**Pull request by FrancescAlted**
This is an attempt to simplify mapping.py as much as possible and promote it as a first-class module.  Not ready for merging yet.

### CODE REVIEW

**Comment by FrancescAlted** [comment]
This should be ready for merging.  @jfburkhart would you like to review this one?

File: reflexible/mapping.py [hunk 1]
```diff
@@ -0,0 +1,1942 @@
...
+                "       </LineString>\n"
+                " </Placemark>\n")
+            file.close()
+
+
+def map_regions(map_region='default', map_par=None, fig_par=None):
+    """
+    This is a core function, used throughout this module. It is called
+    by the :func:`get_FIGURE` function. If you want to create a new
+    region, this is where it should be added. Follow the protocol for the
```
Comment (jfburkhart) [inline reflexible/mapping.py]: Need to change the docstring to reflect the new environment variable

Comment (FrancescAlted) [inline reflexible/mapping.py]: Addressed in 98da4cb

**Comment by FrancescAlted** [comment]
@jfburkhart , in f3cde17 I have tried to clean-up the code of plotting.py a bit.  In the process, I noticed that there are several places where we could be using uninitialized variables.  I added  comments with the "ERROR:" label on them.  Could you have a look at them and tell me which could be proper defaults?

**Comment by FrancescAlted** [comment]
Added an issue for the initialized variables: https://github.com/spectraphilic/reflexible/issues/13

**Comment by FrancescAlted** [comment]
I think this PR is already including more tasks than originally intended, so merging.

**Pull request merged**